File: TGPR.Database/TGPR.Database.Common/Models/Applications/ApplicationTemplateModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace TGPR.Database.Common.Models.Applications
{
    public class ApplicationTemplateModel
    {
        public int ApplicationTemplateId { get; set; }

        [Required]
        public int ApplicationTypeId { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; }

        [Required]
        public bool Active { get; set; }

        [Required]
        public bool Deleted { get; set; }

        public DateTime CreatedOn { get; set; }

        public DateTime? UpdatedOn { get; set; }

        public ApplicationTypeModel Type { get; set; }

        public ICollection<ApplicationCategoryModel> Categories { get; set; }
    }
}

File: TGPR.Database/TGPR.Database.Api/client/src/app/_models/administration/applications/application-template.model.ts
import {ApplicationTypeModel} from "../../applications/application-type.model";
import {ApplicationCategoryModel} from "../../applications/application-category.model";

export class ApplicationTemplateModel {
    ApplicationTemplateId?: number;
    Name: string;
    ApplicationTypeId: number;
    Type: ApplicationTypeModel;
    Active: boolean;
    Deleted: boolean;
    Categories: ApplicationCategoryModel[];
    CreatedOn: Date;
    UpdatedOn: Date;
}
